### PR TITLE
2022 08 serialization

### PR DESCRIPTION
**Pull request by clipclapclop**
These changes change how `PublicKey`, `PublicKeySet`, and `PublicKeyShare` are serialized (and deserialized) so that, when serialized with `serde_json::to_string`, they're represented as hex Strings or an object containing a list of hex Strings. The de/serialization of everything else has been left unchanged. This is meant to address https://github.com/fedimint/minimint/issues/109.

### CODE REVIEW

**Comment by justinmoon** [comment]
Nice work. This does seem to fix https://github.com/fedimint/minimint/issues/109. Without this change the public key is serialized as array of bytes, and with this change it's serialized as hex. I'll defer to @elsirion about the implementation, but in terms of functionality it looks right.

**Review by elsirion — APPROVED** [review]
To be more idiomatic we'd want to serialize as bytes if the encoding format isn't human readable, but this is good enough for our use cases. I'm also not sure if the `CurveWrap` struct is strictly needed, but LGTM.

**Comment by justinmoon** [comment]
We still need to update fedimint to actually use this

**Comment by clipclapclop** [comment]
I can create a pull request shortly.

**Pull request merged**
